pkg/plugin/dumpstate.go

```diff
@@ -0,0 +1,255 @@
+package plugin
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	"github.com/neondatabase/autoscaling/pkg/api"
+	"github.com/neondatabase/autoscaling/pkg/util"
+)
+
+type dumpStateConfig struct {
+	Port           uint16 `json:"port"`
+	TimeoutSeconds uint   `json:"timeoutSeconds"`
+}
+
+func (c *dumpStateConfig) validate() (string, error) {
+	if c.Port == 0 {
+		return "port", errors.New("value must be > 0")
+	} else if c.TimeoutSeconds == 0 {
+		return "timeoutSeconds", errors.New("value must be > 0")
+	}
+
+	return "", nil
+}
+
+type stateDump struct {
+	Stopped   bool            `json:"stopped"`
+	BuildInfo util.BuildInfo  `json:"buildInfo"`
+	State     pluginStateDump `json:"state"`
+}
+
+func (p *AutoscaleEnforcer) startDumpStateServer(shutdownCtx context.Context) error {
+	// Manually start the TCP listener so we can minimize errors in the background thread.
+	addr := net.TCPAddr{IP: net.IPv4zero, Port: int(p.state.conf.DumpState.Port)}
+	listener, err := net.ListenTCP("tcp", &addr)
+	if err != nil {
+		return fmt.Errorf("Error binding to %v", addr)
+	}
+
+	go func() {
```

Comment:
I read this function a few times to make sure that it couldn't panic... I think we're safe but the unchecked panic here makes me nervous for people to edit this in the future.

Reply:
Not sure what the suggestion is. OTOH there's plenty of other places we aren't catching panics, either